src/components/core/GeoChart.tsx

```diff
@@ -0,0 +1,239 @@
+    "#aa8f00",
+    "#3498db",
+    "#3477db",
+    "#2277db",
+    "#bb77db",
+    "#0077db",
+    "#3aa7db",
+];
+const MAP_PIN_SIZES = [8, 9, 12, 16, 25];
+
+const MAPBOX_ACCESS_TOKEN =
+    "pk.eyJ1IjoiaW1udXR6IiwiYSI6ImNrMHAxY2UxZzBnc2EzZG11YmVhd2dubG0ifQ.bUTN7ceAHq6kVooe3MKgqg";
+const MAPBOX_STYLE = "mapbox://styles/mapbox/streets-v11";
+mapboxgl.accessToken = MAPBOX_ACCESS_TOKEN;
+
+export interface IGeoChartProps extends ICommonChartProps, IDataSourceProviderInjectedProps {
+    // cancelPagePromises?: () => void;
+    config?: {}; // IGeoChartConfig;
+}
+
+export interface IGeoChartState {
+    isLoading: boolean;
+    lat: number;
+    lng: number;
+    zoom: number;
+}
+
+export type IGeoChartInnerProps = IGeoChartProps &
+    ILoadingInjectedProps &
+    IDataSourceProviderInjectedProps &
+    WrappedComponentProps;
+
+/**
+ * Geo Chart react component
+ */
+export class GeoChartInner extends BaseVisualization<IGeoChartInnerProps, IGeoChartState> {
+    public static defaultProps: Partial<IGeoChartInnerProps> = {
+        ...commonDefaultProps,
+        onDataTooLarge: noop,
+        config: {},
+    };
+
+    private map: mapboxgl.Map;
+    private mapContainerRef: HTMLDivElement;
+    private mapDataSource: IGeoDataSourceFeatures;
+    private colorMax: number;
+    private colorMin: number;
+    private sizeMax: number;
+    private sizeMin: number;
+
+    constructor(props: IGeoChartInnerProps) {
+        super(props);
+
+        this.state = {
+            isLoading: true,
+            lat: DEFAULT_LOCATION.LAT,
+            lng: DEFAULT_LOCATION.LNG,
+            zoom: DEFAULT_LOCATION.ZOOM,
+        };
+    }
+
+    public componentDidMount = () => {
+        this.loadMap();
+        this.createDataSource();
+    };
+
+    public renderVisualization = () => {
+        const style: React.CSSProperties = {};
+        return (
+            <div className="gd-geo-component" style={style}>
+                <div className="mapContainer" ref={this.setMapContainerRef} />]{this.renderLoadingOverlay()}
+            </div>
+        );
+    };
+
+    private renderLoadingOverlay = (): React.ReactNode => {
+        if (this.state.isLoading) {
+            return <LoadingComponent />;
+        }
+        return null;
+    };
+
+    //
+    // getters / setters / manipulators
+    //
+
+    private setMapContainerRef = (container: HTMLDivElement): void => {
+        this.mapContainerRef = container;
+    };
+
+    private loadMap = (): void => {
+        const { lat, lng, zoom } = this.state;
+        const map = new mapboxgl.Map({
+            container: this.mapContainerRef,
+            style: MAPBOX_STYLE,
+            center: [lng, lat],
+            zoom,
+        });
+        map.on("move", this.mapMoved);
+        this.map = map;
+    };
+
+    //
+    // working with data source
+    //
+
+    private createDataSource = (): void => {
+        createDataSource(this.props.resultSpec, this.props.getPage).then(this.dataSourceCreated);
```

Comment:
Please handle the loading data outside of `CoreGeoChart`
`CoreGeoChart` will receive the `execution` object